frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala

```diff
@@ -1,81 +1,159 @@
 package akkaviz.frontend.components
 
-import akkaviz.frontend.DOMGlobalScope
-import akkaviz.frontend.components.GraphView.AddLink
-import org.scalajs.dom.console
-import org.scalajs.dom.html.Element
+import akkaviz.frontend.vis.NetworkOptions
+import akkaviz.frontend.{ScheduledQueue, vis}
+import org.scalajs.dom._
 import rx.Var
 
-import scala.concurrent.duration.{FiniteDuration, _}
 import scala.scalajs.js
-import scala.scalajs.js.timers.SetTimeoutHandle
+import scala.scalajs.js.JSConverters._
+import scala.scalajs.js.|._
 
-class GraphView(showUnconnected: Var[Boolean]) extends Component {
+class GraphView(showUnconnected: Var[Boolean], actorSelectionToggler: (String) => Unit) extends Component {
 
-  override def render: Element = {
+  private[this] lazy val visibleNodes = js.Dictionary[Unit]()
+  private[this] lazy val networkNodes = new vis.DataSet[vis.Node]()
+  private[this] lazy val networkEdges = new vis.DataSet[vis.Edge]()
+  private[this] var network: js.UndefOr[vis.Network] = js.undefined
 
-    //FIXME: port initialization of graph from JS to scala and do it here
-    js.undefined.asInstanceOf
-  }
+  override def attach(parent: Element): Unit = {
+    network.foreach(_.destroy())
 
-  private val graph = DOMGlobalScope.graph
-  private val nodes = js.Dictionary[js.Dictionary[js.Any]]()
-  private val connectedNodes = js.Dictionary[Unit]()
-  private val createdLinks = js.Dictionary[Unit]()
+    val data = vis.NetworkData(networkNodes, networkEdges)
+    val n = new vis.Network(parent, data, graphSettings)
+    n.onDoubleClick {
+      (event: vis.ClickEvent) =>
+        event.nodes.foreach(actorSelectionToggler)
+    }
 
-  private val updateFrequency: FiniteDuration = 500.millis
+    network = n
+  }
 
-  private var scheduler: js.UndefOr[SetTimeoutHandle] = js.undefined
+  private def graphSettings: NetworkOptions = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [graphSettings should have a smaller scope by using private[this]](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736317/issues/source?bid=3121282&fileBranchId=3158877#l32)